drivers/soundwire/cadence_master.c

```diff
@@ -868,6 +869,20 @@ int sdw_cdns_enable_interrupt(struct sdw_cdns *cdns, bool state)
 		mask = interrupt_mask;
 
 update_masks:
+	/* clear slave interrupt status before enabling interrupt */
+	if (state) {
+		slave_state0 = cdns_readl(cdns, CDNS_MCP_SLAVE_INTSTAT0);
+		slave_state1 = cdns_readl(cdns, CDNS_MCP_SLAVE_INTSTAT1);
+
+		if (slave_state0)
+			cdns_writel(cdns, CDNS_MCP_SLAVE_INTSTAT0,
+				    slave_state0);
+
+		if (slave_state1)
+			cdns_writel(cdns, CDNS_MCP_SLAVE_INTSTAT1,
+				    slave_state1);
+	}
+
```

Comment:
you can simplify by using a single variable and unconditional writes. it's a clear on write1, so there's no harm in writing zero, and it's not saving much time anyways.

Reply:
thanks, I will refine it.